### PR TITLE
raftstore: bcast commit for urgent requests

### LINKED ISSUE #3578: skip_bcast_commit will delay follower's apply process
## Bug Report

**What version of Rust are you using?**
rustc 1.29.0-nightly (4f3c7a472 2018-07-17)

**What operating system and CPU are you using?**
Centos 7.3

**What did you do?**
Writing data continuously.

**What did you expect to see?**
Leader will be elected as soon as split finished.

**What did you see instead?**
I find that if region split happens, the new generated region can't elect a leader for about 2 seconds. After some investigation, i found that two followers applied split command 2 seconds after it has applied in leader. If i turn off `skip_bcast_commit`, this delay will go away.

### PR DESCRIPTION
**Pull request by BusyJay**
TiKV disable broadcast commit by default to reduce messages, which lead
to good overall performance. This option can also lead to delay apply on
all follower nodes. Sometimes it's unexpected for urgent requests like
split or similiar. This PR enables broadcast commit for those requests
temporary.

Close #3578.

### CODE REVIEW

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -1062,6 +1064,10 @@ impl Peer {
             }
             if !committed_entries.is_empty() {
                 self.last_applying_idx = committed_entries.last().unwrap().get_index();
+                if self.last_applying_idx >= self.last_urgent_proposal_idx {
+                    self.raft_group.skip_bcast_commit(true);
```
Comment (overvenus) [inline src/raftstore/store/peer.rs]: Please add some comments.

File: Cargo.toml [hunk 2]
```diff
@@ -82,6 +82,9 @@ num = "0.2.0"
 hex = "0.3"
 rust-crypto = "^0.2"
 
+[replace]
+"raft:0.3.1" = { git = "https://github.com/pingcap/raft-rs.git" }
```
[the remark this answers is not in the document]
Comment (Hoverbear) [reply Cargo.toml]: We can cut a 0.3.2 :)

**Comment by BusyJay** [comment]
@Hoverbear I'm OK with either way. :bear:

**Comment by Hoverbear** [comment]
Hmm, looking at https://github.com/pingcap/raft-rs/milestone/4?closed=1 it might make more sense to cut 0.4.0 and assign pending issues to 0.5.0. It's already got 5 more PRs closed than any other of the releases.

I'll go ahead and do that tomorrow. I was already worrying about 0.4.0 being too big (there are pending API changes, etc).

**Comment by Hoverbear** [comment]
@siddontang I'm just waiting on these PRs to merge: https://github.com/pingcap/raft-rs/milestone/4

Should be able to do very soon.

**Pull request merged**